ipatests/test_webui/ui_driver.py

```diff
@@ -1700,34 +1751,86 @@ def get_t_vals(t):
             # add multiple at once and test table delete button
             self.add_table_associations(table, keys, delete=True)
 
-    def add_sshkey_to_user(self, user, ssh_key):
+    def add_sshkey_to_record(self, ssh_keys, pkey, navigate=False,
+                             save=True):
         """
-        Add ssh public key to particular user
+        Add ssh public key to particular record
 
-        user (str): user to add the key to
-        ssh_key (str): public ssh key
+        ssh_keys (list): public ssh key(s)
+        pkey (str): user/host/idview to add the key to
+        navigate (bool): whether we should navigate to record
+        save (bool): whether we should click save after adding a key
         """
-        self.navigate_to_entity('user')
-        self.navigate_to_record(user)
 
-        ssh_pub = 'div[name="ipasshpubkey"] button[name="add"]'
-        self.find(ssh_pub, By.CSS_SELECTOR).click()
-        self.wait()
-        self.driver.switch_to.active_element.send_keys(ssh_key)
-        self.dialog_button_click('update')
-        self.facet_button_click('save')
+        if type(ssh_keys) is not list:
+            ssh_keys = [ssh_keys]
+
+        if navigate:
+            self.navigate_to_record(pkey)
+
+        for key in ssh_keys:
+            s_add = 'div[name="ipasshpubkey"] button[name="add"]'
+            ssh_add_btn = self.find(s_add, By.CSS_SELECTOR, strict=True)
+            ssh_add_btn.click()
+            self.wait()
+            s_text_area = 'textarea.certificate'
+            text_area = self.find(s_text_area, By.CSS_SELECTOR, strict=True)
+            text_area.send_keys(key)
+            self.wait()
+            self.dialog_button_click('update')
+
+        # sometimes we do not want to save e.g. in order to test undo buttons
+        if save:
+            self.facet_button_click('save')
 
-    def delete_user_sshkey(self, user):
+    def delete_record_sshkeys(self, pkey, navigate=False):
         """
-        Delete ssh public key of particular user
+        Delete all ssh public keys of particular record
+
+        pkey (str): user/host/idview to add the key to
+        navigate (bool): whether we should navigate to record
         """
-        self.navigate_to_entity('user')
-        self.navigate_to_record(user)
+
+        if navigate:
+            self.navigate_to_record(pkey)
```

Comment:
I'm not sure about this navigation change, I thought/expected that the change will be:

```
def delete_record_sshkeys(self, entity, pkey, navigate=False):
    ....
    if navigate:
            self.navigate_to_entity(entity)
            self.navigate_to_record(pkey)
```

to be consistent with other navigation code, otherwise seems OK. Don't know if to block the review on that.